src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlConnection.cs

```diff
@@ -26,6 +31,33 @@ namespace Microsoft.CodeAnalysis.SQLite.v2.Interop
     /// </summary>
     internal class SqlConnection
     {
+        // Cached utf8 (and null terminated) versions of the common strings we need to pass to sqlite.  Used to prevent
+        // having to convert these names to/from utf16 to utf8 on every call.  Sqlite requires these be null terminated.
+
+        private static readonly byte[] s_mainNameWithTrailingZero = GetUtf8BytesWithTrailingZero(Database.Main.GetName());
```

Comment:
❔ Can we not hold `ImmutableArray<byte>` here? We'll still be able to get a `ReadOnlySpan<byte>` from that.

Reply:
primary issue is that Sqlite pcl doesn't take a ROS here, they take a `byte*+length`.  I have an outstanding PR with them to allow a ROS here, which would allow us to make this immutable.